salt/_modules/azurearm_network.py

```diff
@@ -2749,3 +2749,1408 @@ def route_tables_list_all(**kwargs):
+            virtual_network_gateway_name=name,
+            vpnclient_ipsec_params=paramsmodel,
+            **kwargs
+        )
+
+        params_result = params.result()
+        result = params_result.as_dict()
+    except CloudError as exc:
+        __utils__['azurearm.log_cloud_error']('network', str(exc), **kwargs)
+        result = {'error': str(exc)}
+
+    return result
+
+
+def virtual_network_gateway_get_vpnclient_ipsec_parameters(name, resource_group, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Gets information about the vpnclient ipsec policy for P2S client of virtual network
+        gateway in the specified resource group through Network resource provider.
+
+    :param name: The virtual network gateway name.
+
+    :param resource_group: The name of the resource group.
+
+    CLI Example:
+    .. code-block:: bash
+    
+        salt-call azurearm_network.virtual_network_gateway_get_vpnclient_ipsec_parameters test_name test_group
+
+    '''
+    netconn = __utils__['azurearm.get_client']('network', **kwargs)
+    try:
+        policy = netconn.virtual_network_gateways.get_vpnclient_ipsec_parameters(
+            resource_group_name=resource_group,
+            virtual_network_gateway_name=name
+        )
+
+        policy_result = policy.result()
+        result = policy_result.as_dict()
+    except CloudError as exc:
+        __utils__['azurearm.log_cloud_error']('network', str(exc), **kwargs)
+        result = {'error': str(exc)}
+
+    return result
+
+
+def virtual_network_gateway_vpn_device_configuration_script(name, resource_group, vendor, device_family, 
+							    firmware_version, **kwargs):
+    '''
+    .. versionadded:: Sodium
+
+    Gets a xml format representation for vpn device configuration script.
+
+    :param name: The name of the virtual network gateway connection for which the configuration
+        script is generated.
+
+    :param resource_group: The name of the resource group.
+
+    :param vendor: The vendor for the vpn device.
+
+    :param device_family: The device family for the vpn device.
+
+    :param firmware_version: The firmware version for the vpn device. 
+
+    CLI Example:
+    .. code-block:: bash
+
+        salt-call azurearm_network.virtual_network_gateway_vpn_device_configuration_script test_name test_group \ 
+	          test_vendor test_device_fam test_version
+
+    '''
+    netconn = __utils__['azurearm.get_client']('network', **kwargs)
+
+    try:
+        scriptmodel = __utils__['azurearm.create_object_model'](
+            'network',
+            'VpnDeviceScriptParameters',
+            vendor=vendor,
+            device_family=device_family,
+	    firmware_version=firmware_version,
+        )
+    except TypeError as exc:
+        result = {'error': 'The object model could not be built. ({0})'.format(str(exc))}
+        return result
+
+    try:
+        script = netconn.virtual_network_gateways.vpn_device_configuration_script(
```

Comment:
does this spit out the script text or a file?

Reply:
The XML format representation for the vpn device configuration script